registries/defaultregistry/default.go

```diff
@@ -188,25 +190,54 @@ func (reg *DefaultRegistry) GetV2Token(client *http.Client, url string) (*common
 //multiple tags on a single image will be sent as a comma separated string
 //e.g ["latest,v3" ,"v2", "v1"]
 func (reg *DefaultRegistry) GetLatestTags(repoName string, depth int, options ...remote.Option) ([]string, error) {
-
+	type imageInfo struct {
+		created time.Time
+		digest  string
+		tag     string
+		err     error
+	}
 	tagsInfos := tagsInfo{}
+	wg := sync.WaitGroup{}
+	ctx, cancelCtx := context.WithCancel(context.Background())
+	defer cancelCtx()
 	for tagsPage, nextPage, err := reg.This.List(repoName, common.MakePagination(reg.This.GetMaxPageSize()), options...); ; tagsPage, nextPage, err = reg.This.List(repoName, *nextPage, options...) {
 		if err != nil {
 			return nil, err
 		}
+		//if depth is one (default) and latest tag found no need to continue
+		if depth == 1 && slices.Contains(tagsPage, "latest") {
+			return []string{"latest"}, nil
+		}
+		ch := make(chan imageInfo, len(tagsPage))
```

Comment:
We must limit the number of goroutines to some constant.
30 is more than enough IMHO